cmd/jujud/agent/machine.go

```diff
@@ -716,21 +657,22 @@ func (a *MachineAgent) newRestoreStateWatcherWorker(st *state.State) (worker.Wor
 	return jworker.NewSimpleWorker(rWorker), nil
 }
 
+func (a *MachineAgent) getRestoreStatus() state.RestoreStatus {
+	a.restoreStatusMu.Lock()
+	defer a.restoreStatusMu.Unlock()
+	return a.restoreStatus
```

Comment:
presumably this is already getting set somewhere else correctly?

Reply:
a.restoreStatus is set by the a.restoreChanged method, which is called by restoreStateWatcher